x/ecocredit/server/testsuite/queries.go

```diff
@@ -289,3 +289,348 @@ func (s *IntegrationTestSuite) TestCreditTypeQuery() {
+		expectErr bool
+		errMsg    string
+	}{
+		{
+			"nil request",
+			nil,
+			true,
+			"empty request",
+		},
+		{
+			"valid request",
+			&ecocredit.QuerySellOrdersRequest{},
+			false,
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			_, err := s.queryClient.SellOrders(s.ctx, tc.request)
+			if tc.expectErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.errMsg)
+			} else {
+				require.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestSellOrdersByAddressQuery() {
+	require := s.Require()
+
+	testCases := []struct {
+		name      string
+		request   *ecocredit.QuerySellOrdersByAddressRequest
+		expectErr bool
+		errMsg    string
+	}{
+		{
+			"nil request",
+			nil,
+			true,
+			"empty request",
+		},
+		{
+			"empty request",
+			&ecocredit.QuerySellOrdersByAddressRequest{},
+			true,
+			"empty address string is not allowed",
+		},
+		{
+			"valid request",
+			&ecocredit.QuerySellOrdersByAddressRequest{
+				Address:    s.signers[3].String(),
+			},
+			false,
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			_, err := s.queryClient.SellOrdersByAddress(s.ctx, tc.request)
+			if tc.expectErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.errMsg)
+			} else {
+				require.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestSellOrdersByBatchDenomQuery() {
+	require := s.Require()
+
+	testCases := []struct {
+		name      string
+		request   *ecocredit.QuerySellOrdersByBatchDenomRequest
+		expectErr bool
+		errMsg    string
+	}{
+		{
+			"nil request",
+			nil,
+			true,
+			"empty request",
+		},
+		{
+			"empty request",
+			&ecocredit.QuerySellOrdersByBatchDenomRequest{},
+			true,
+			"invalid denom",
+		},
+		{
+			"valid request",
+			&ecocredit.QuerySellOrdersByBatchDenomRequest{
+				BatchDenom: "A00-00000000-00000000-000",
+			},
+			false,
+			"",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			_, err := s.queryClient.SellOrdersByBatchDenom(s.ctx, tc.request)
+			if tc.expectErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.errMsg)
+			} else {
+				require.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestBuyOrderQuery() {
+	require := s.Require()
+
+	testCases := []struct {
+		name      string
+		request   *ecocredit.QueryBuyOrderRequest
+		expectErr bool
+		errMsg    string
+	}{
+		{
+			"nil request",
+			nil,
+			true,
+			"empty request",
+		},
+		{
+			"empty request",
+			&ecocredit.QueryBuyOrderRequest{},
+			true,
+			"not found",
+		},
+		{
+			"invalid order id",
+			&ecocredit.QueryBuyOrderRequest{
+				BuyOrderId: 99,
+			},
+			true,
+			"not found",
+		},
+		// TODO: filtered buy orders required #623
+		//{
+		//	"valid request",
+		//	&ecocredit.QueryBuyOrderRequest{
+		//		BuyOrderId: 1,
+		//	},
+		//	false,
+		//	"",
+		//},
```

Comment:
Currently we are not storing direct buy orders and we have not implemented indirect buy orders, so creating a buy order with `Buy` will not store a buy order and we have no buy orders to query.